packages/form-js-viewer/test/spec/import/Importer.spec.js

```diff
@@ -0,0 +1,87 @@
+import {
+  bootstrapForm,
+  getForm,
+  inject
+} from 'test/TestHelper';
+
+import { clone } from 'src/util';
+
+import schema from '../form.json';
+
+describe('Importer', function() {
+
+  beforeEach(bootstrapForm());
+
+  afterEach(function() {
+    getForm().destroy();
+  });
+
+
+  it('should import without errors', inject(async function(form, formFieldRegistry) {
+
+    // given
+    const data = {
+      creditor: 'John Doe Company',
+      amount: 456,
+      invoiceNumber: 'C-123',
+      approved: true,
+      approvedBy: 'John Doe',
+      product: 'camunda-cloud',
+      language: 'english'
+    };
+
+    // when
+    const { err, warnings } = await form.importSchema(schema, data);
+
+    // then
+    expect(err).not.to.exist;
+    expect(warnings).to.be.empty;
+
+    expect(formFieldRegistry.size).to.equal(11);
+  }));
+
+
+  it('should error if form field of type not supported', inject(async function(form) {
+
+    // given
+    const error = clone(schema);
+
+    error.components.push({
+      type: 'foo'
+    });
+
+    // when
+    try {
+      await form.importSchema(error);
+    } catch (err) {
+
+      // then
+      expect(err).to.exist;
+      expect(err.message).to.eql('form field of type <foo> not supported');
+    }
+  }));
+
+
+  it('should error if form field with key already exists', inject(async function(form) {
+
+    // given
+    const error = clone(schema);
+
+    error.components.push({
+      type: 'textfield',
+      key: 'creditor'
+    });
+
+    // when
+    // when
+    try {
+      await form.importSchema(error);
+    } catch (err) {
+
+      // then
+      expect(err).to.exist;
+      expect(err.message).to.eql('form field with key <creditor> already exists');
+    }
+  }));
+
```

Comment:
Can we also add a corrupt JSON error, i.e. import binary or import an incomplete `SELECT` element, i.e. missing options?

Reply:
I've added 2 tests for both broken JSON and incomplete form fields (cf. https://github.com/bpmn-io/form-js/pull/83/files#diff-023dda47ee2cfd7020670740515f71a05a093186786bce65036fce5dba6ecd8fR94). The error handling can be improved. Furthermore, we cannot catch errors due to incomplete form fields at the moment as they will only occur during rendering.